src/components/text-editor/link-menu/limel-text-editor-link-menu.tsx

```diff
@@ -0,0 +1,150 @@
+import translate from '../../../global/translations';
+
+/**
+ * This component is a menu for editing a link in the text editor.
+ * It allows the user to input a title of a link and a link itself.
+ */
+
+@Component({
+    tag: 'limel-text-editor-link-menu',
+    shadow: true,
+    styleUrl: 'limel-text-editor-link-menu.scss',
+})
+export class TextEditorLinkMenu {
+    /**
+     * The link
+     */
+    @Prop()
+    public link: EditorTextLink;
+
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop({ reflect: true })
+    public language: Languages = 'en';
+
+    /**
+     * Open state of the dialog
+     */
+    @Prop()
+    public isOpen: boolean = false;
+
+    @State()
+    private invalidLink: boolean = false;
+
+    /**
+     * Emitted when the menu is closed from inside the component.
+     * (*Not* emitted when the consumer sets the `open`-property to `false`.)
+     */
+    @Event()
+    private cancel: EventEmitter<void>;
+
+    /**
+     * Emitted when the menu is saved from inside the component.
+     */
+    @Event()
+    private save: EventEmitter<void>;
+
+    /**
+     * Emitted when the user inputs new values for the link
+     */
+    @Event()
+    private linkChange: EventEmitter<EditorTextLink>;
+
+    public render() {
+        return [
+            <limel-input-field
+                label="Text"
+                value={this.link?.text || ''}
+                leadingIcon="text_cursor"
+                onChange={this.handleLinkTitleChange}
+            />,
+            <limel-input-field
+                label="Link"
+                value={this.link?.href || ''}
+                leadingIcon="-lime-text-link"
+                invalid={this.invalidLink}
+                onChange={this.handleLinkValueChange}
+            />,
+            <div class="actions">
+                <limel-button
+                    label={this.getTranslation('cancel')}
+                    onClick={this.handleCancel}
+                    slot="button"
+                />
+                <limel-button
+                    primary={true}
+                    label={this.getTranslation('save')}
+                    disabled={!this.link?.href || this.invalidLink}
+                    onClick={this.handleSave}
+                    slot="button"
+                />
+            </div>,
+        ];
+    }
+
+    private getTranslation(key: string) {
+        return translate.get(`${key}`, this.language);
+    }
+
+    private validateLink = (href: string): boolean => {
+        if (this.isValidLink(href)) {
+            return true;
+        }
+
+        // If the link doesn't have a scheme, add 'http://' and validate again
+        if (!/^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(href)) {
+            const formattedLink = 'http://' + href;
+            if (this.isValidLink(formattedLink)) {
+                this.link = { ...this.link, href: formattedLink };
```

Comment:
Ah, hang on! We're actually _updating_ the input value here?

Should we really do that? If we do, I think we should use `https` rather than `http`.

Reply:
I did add that and I'm not sure if this is what we want. If the user would insert a link to `www.google.com` they would expect that link to be absolute and not relative, I would assume.

So, `http://www.google.com`
Rather than, `https://lundalogik.github.io/lime-elements/versions/PR-2994/www.google.com`

But if they enter `/client/object/company/1033` the user might actually expect the link to be relative.

I don't know what we should support here and how clever we should try to be.

Reply:
I'm not so sure that we are ultimately prepending the schema to the link if missing. This is only the link validation method.

Reply:
I don't think we should update the input value here. It should only handle link validation, nothing more. I'll change this in a fixup.

Reply:
I've removed the update to the link as it's only used for validation. We also might not necessarily want to prepend a protocol to what a user might paste as a link?